TLM/TLM/Util/PriorityRoad.cs

```diff
@@ -9,18 +9,86 @@ namespace TrafficManager.Util {
     using UnityEngine;
     using static TrafficManager.Util.SegmentTraverser;
     using State;
-    using System.Linq;
-    using static Util.Shortcuts;
+    using static TrafficManager.Util.Shortcuts;
+    using static TrafficManager.UI.SubTools.PrioritySignsTool;
+    using TrafficManager.API.Manager;
 
     public static class PriorityRoad {
+        public static void FixPrioritySigns(PrioritySignsMassEditMode massEditMode, List<ushort> segmentList) {
+            if (segmentList == null || segmentList.Count == 0) {
+                return;
+            }
+
+            var primaryPrioType = PriorityType.None;
+            var secondaryPrioType = PriorityType.None;
+
+            switch (massEditMode) {
+                case PrioritySignsMassEditMode.MainYield: {
+                        primaryPrioType = PriorityType.Main;
+                        secondaryPrioType = PriorityType.Yield;
+                        break;
+                    }
+
+                case PrioritySignsMassEditMode.MainStop: {
+                        primaryPrioType = PriorityType.Main;
+                        secondaryPrioType = PriorityType.Stop;
+                        break;
+                    }
+
+                case PrioritySignsMassEditMode.YieldMain: {
+                        primaryPrioType = PriorityType.Yield;
+                        secondaryPrioType = PriorityType.Main;
+                        break;
+                    }
+
+                case PrioritySignsMassEditMode.StopMain: {
+                        primaryPrioType = PriorityType.Stop;
+                        secondaryPrioType = PriorityType.Main;
+                        break;
+                    }
+            }
+
+            IExtSegmentEndManager segEndMan = Constants.ManagerFactory.ExtSegmentEndManager;
+
+            void ApplyPrioritySigns(ushort segmentId, bool startNode) {
+                ushort nodeId = netService.GetSegmentNodeId(
+                    segmentId,
+                    startNode);
+
+                TrafficPriorityManager.Instance.SetPrioritySign(
+                    segmentId,
+                    startNode,
+                    primaryPrioType);
+
+                for (int i = 0; i < 8; ++i) {
+                    ushort otherSegmentId = nodeId.ToNode().GetSegment(i);
+                    if (otherSegmentId == 0 ||
+                        otherSegmentId == segmentId ||
+                        segmentList.Contains(otherSegmentId)) {
+                        continue;
+                    }
+
+                    TrafficPriorityManager.Instance.SetPrioritySign(
+                        otherSegmentId,
+                        (bool)netService.IsStartNode(otherSegmentId, nodeId),
+                        secondaryPrioType);
+                }
+            }
+
+            foreach(ushort segId in segmentList) {
+                foreach(bool bStartNode in Constants.ALL_BOOL) {
+                    ApplyPrioritySigns(segId, bStartNode);
+                }
```

Comment:
I believe calling it twice with false and true will look simpler than a loop.
Up to you to judge, but simpler is better.

Reply:
you are right. this is remnants of old code.